### PR TITLE
Re-enable reason in CI

### PR DESCRIPTION
**Pull request by rgrinberg**
Reason's dependency profile is more reasonable now. We can attempt to re-enable it

cc @jordwalke

### CODE REVIEW

File: test/blackbox-tests/test-cases/reason/jbuild [hunk 1]
```diff
@@ -4,31 +4,31 @@
 (library
  ((name rlib)
   (public_name rlib)
-  (modules bar cppome foo hello pped)
+  (modules (bar cppome foo hello pped))
```
Comment (hcarty) [inline test/blackbox-tests/test-cases/reason/jbuild]: Out of curiosity - why the change from `dune` to `jbuilder`?  Is Reason supported under Dune 1.0.0?

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/reason/jbuild]: Yeah, it's supported. I don't  quite remember as the PR was sitting around. I think it's because of a ppx reason. I'll try again.

File: test/blackbox-tests/test-cases/reason/run.t [hunk 2]
```diff
@@ -1,94 +1,109 @@
   $ dune build @runtest @install-file --display short
+  Info: creating file dune-project with this contents: (lang dune 1.0)
```
Comment (diml) [inline test/blackbox-tests/test-cases/reason/run.t]: We should create these dune-project files in tests. This way we don't have to update all the tests at once when we'll change something.

File: test/blackbox-tests/test-cases/reason/jbuild (hunk 1 above)
Comment (hcarty) [reply test/blackbox-tests/test-cases/reason/jbuild]: Thanks - just checking before I dive into testing dune against a local mixed OCaml/Reason codebase.
